codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/DafnyLocalServiceGenerator.java

```diff
@@ -633,30 +740,29 @@ void generateReferencedResources(GenerationContext context) {
                   .expectShape(resource, ResourceShape.class)
                   .getOperations()
                   .forEach(operation -> {
-                    var operationShape = model.expectShape(
+                    final var operationShape = model.expectShape(
                       operation,
                       OperationShape.class
                     );
-                    final var input = model
-                        .expectShape(operationShape.getInputShape())
-                        .hasTrait(UnitTypeTrait.class)
-                      ? ""
-                      : operationShape.getInputShape().getName();
-                    final var output = model
-                        .expectShape(operationShape.getOutputShape())
-                        .hasTrait(UnitTypeTrait.class)
-                      ? ""
-                      : "*%s,".formatted(
-                          operationShape.getOutputShape().getName()
-                        );
-
+                    final var input =
+                      GoCodegenUtils.getOperationalShapeInputName(
+                        model,
+                        operationShape,
+                        symbolProvider
+                      );
+                    final var output =
+                      GoCodegenUtils.getOperationalShapeOutputName(
+                        model,
+                        operationShape,
+                        symbolProvider
+                      );
                     w.write(
                       """
                       $L($L) ($L error)
                       """,
                       operationShape.getId().getName(),
                       input,
-                      output
+                      output.equals("") ? "" : output.concat(",")
```

Comment:
Nit: add a comment. 
I _think_ this is for a void/Unit response.
Or make the ternary's switch be `hasTrait(UnitTypeTrait.class)`.
That would be easier to follow, but more typing/characters.

Either is better than making me scratch my head for a minute or two...

Reply:
Added a comment with isEmpty() check